File: src/main/kotlin/com/digisphere/setup/auth/dto/LoginDto.kt
package com.digisphere.setup.auth.dto

import jakarta.validation.constraints.NotBlank

data class LoginRequest(
    @field:NotBlank
    val email: String,

    @field:NotBlank
    val password: String,
)

data class RefreshTokenRequest(
    @field:NotBlank
    val refreshToken: String
)

data class LoginResponse(
    val accessToken: String,
    val refreshToken: String,
)

data class TokenResponse(
    val accessToken: String
)

File: mobile/modules/auth/dto/login.dto.ts
export interface ILoginRequest {
    email?: string,
    password?: string,
}

export interface ILoginResponse {
    token?: string,
    refreshToken?: string
}

export class LoginRequest implements ILoginRequest {
    public readonly email?: string;
    public readonly password?: string;
}

export class LoginResponse implements ILoginResponse {
    public readonly accessToken?: string
    public readonly refreshToken?: string
}